packages/client/tests/functional/_utils/getTestSuiteInfo.ts

```diff
@@ -130,7 +135,7 @@ function getTestSuiteParametersString(configs: Record<string, string>[]) {
         const providerFlavorStr = config.providerFlavor === undefined ? '' : `providerFlavor=${config.providerFlavor},`
         return `relationMode=${config.relationMode},provider=${config.provider},${providerFlavorStr}onUpdate=${config.onUpdate},onDelete=${config.onDelete},id=${config.id}`
       } else {
-        const firstKey = Object.keys(config)[0]
+        const firstKey = Object.keys(config)[0] // ! TODO this can actually produce incorrect tests and break type checks ! \\ Replace with hash
```

Comment:
Todo for tech debt?

Reply:
Does the comment says it risks producing wrong results because of Object.keys relying on an unstable order for its keys? And, does it break type checks because `Object.keys(config)` could be empty, and thus its 0th entry would be undefined?

As of `ES2020` (supported in Node.js 14.x and onwards), `Object.keys` actually follows a consistent property order. Moreover, `Reflect.ownKeys` enforces this same consistent order even on `ES2019`.

Reply:
This is because of the matrix keys actually, this assumes that every first key in the matrix is a different provider, and that is not always the case.